### PR TITLE
nice try

### PR DESCRIPTION
**Pull request by Kolyo-Coherent**

### CODE REVIEW

File: src/.env [hunk 1]
```diff
@@ -0,0 +1,2 @@
+REACT_APP_AUTH_USERNAME=kolyopeev
+REACT_APP_AUTH_PASSWORD=kx75KiKsnOS8
```
Comment (JuliaShavyrova) [inline src/.env]: now i can use your creds, thanks :)

File: src/actions.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
+import axios from 'axios';
+
+//const apiURL = 'https://pfa.foreca.com/api';
+const accToken =
+  'eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJodHRwOlwvXC9mbnc2LmZvcmVjYS5jb21cL2F1dGhvcml6ZVwvdG9rZW4iLCJpYXQiOjE1MjYzMDAzODAsImV4cCI6MTUyNjMwMzk4MCwibmJmIjoxNTI2MzAwMzgwLCJqdGkiOiJxSXl3WVlQNjc1NkczejBEIiwic3ViIjoibFFHa1Y4Z2pIeGUyZU1ibndUUUs4NktqVTY3RXJlS2htenY1IiwicHJ2IjoiYWY3YTAzOThkZGNiNWE3YTUzN2Q3YzdkMjU2NWEyZjgxZGM4ZTQxNCJ9.V8xg6L9yrY9__VH-jdrL_CqXisEpgcfdUa0NoxlGz0k';
```
Comment (JuliaShavyrova) [inline src/actions.js]: now i can also use your token too :)

about using hardcoded tokens - please add request to get the token, not just use hardcoded one

File: src/proxy.js [hunk 3]
```diff
@@ -0,0 +1,11 @@
+const express = require('express');
+const { createProxyMiddleware } = require('http-proxy-middleware');
+const app = express();
+app.use(
+  '/api',
+  createProxyMiddleware({
+    target: 'https://pfa.foreca.com',
+    changeOrigin: true
```
Comment (JuliaShavyrova) [inline src/proxy.js]: 👏

File: package.json [hunk 4]
```diff
@@ -18,18 +19,23 @@
...
-    "react-redux": "7.2.2",
+    "react-redux": "^7.2.2",
     "react-router-dom": "5.2.0",
-    "redux": "4.0.5",
-    "redux-thunk": "2.3.0",
-    "reselect": "4.0.0"
+    "redux": "^4.0.5",
+    "redux-thunk": "^2.3.0",
+    "reselect": "4.0.0",
+    "xhr2": "^0.2.1"
```
Comment (JuliaShavyrova) [inline package.json]: why do you need this library?

File: proxy.js [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+  axios({
+    method: 'post',
+    url: `https://pfa.foreca.com/api/authorize/token`,
+    data: {
+      user: process.env.REACT_APP_AUTH_USERNAME,
+      password: process.env.REACT_APP_AUTH_PASSWORD
+    }
+  })
+    .then(response => {
+      console.log('API call succeeded!');
```
Comment (JuliaShavyrova) [inline proxy.js:34]: please remove console logs used for debugging (console.error for logging errors is allowed)

File: proxy.js [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+      getApiKey = keyHolder(response.data);
+      res.send({ data: 'Logged in' });
+    })
+    .catch(error => {
+      if (error) console.log('API call returned an error: ', error);
+    });
+});
+
+app.post('/a', (req, res) => {
+  let token2 = getApiKey();
```
Comment (JuliaShavyrova) [inline proxy.js]: variable should be either named more specifically or removed

File: proxy.js [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+  let token = { Authorization: `Bearer ${token2}` };
```
Comment (JuliaShavyrova) [inline proxy.js]: please use const if your variable should not be modified

**Comment by JuliaShavyrova** [comment]
please let me know when your PR will be ready for review

**Comment by Kolyo-Coherent** [comment]
It is since yesterday?

File: src/actions.js (hunk 2 above)
Comment (JuliaShavyrova) [reply src/actions.js]: still see hardcoded token

File: src/index.jsx [hunk 8]
```diff
@@ -1,4 +1,16 @@
...
+import { Provider } from 'react-redux';
+import thunk from 'redux-thunk';
 import App from './App.jsx';
+import postsReucer from './reducer.js';
 
-ReactDOM.render(<App />, document.getElementById('app'));
+//const store = createStore(postsReucer, applyMiddleware(thunk));
+
+ReactDOM.render(
+  //<Provider store={store}>
```
Comment (JuliaShavyrova) [inline src/index.jsx]: please remove commented code

File: package.json (hunk 4 above)
Comment (Kolyo-Coherent) [reply package.json]: I removed it but in other branch.. =]

File: src/index.jsx (hunk 8 above)
Comment (Kolyo-Coherent) [reply src/index.jsx]: updated code to requested changes

**Review by JuliaShavyrova — APPROVED** [review]

**Pull request merged**